vpr/src/noc/xy_routing.h

```diff
@@ -140,7 +140,7 @@ class XYRouting : public NocRouting {
      * that is currently being visited on the FPGA
      * @return RouteDirection The direction to travel next
      */
-    RouteDirection get_direction_to_travel(int sink_router_x_position, int sink_router_y_position, int curr_router_x_position, int curr_router_y_position);
+    static RouteDirection get_direction_to_travel(int sink_router_x_position, int sink_router_y_position, int curr_router_x_position, int curr_router_y_position);
```

Comment:
Is there any reason to make this static?  I guess it doesn't need the calling object and hence can be made static, but if there's no real advantage to making it static (we have an object we can call it with) I'd lean toward keeping it a normal method for simplicity and in case some future change means you need to look something up in the calling object.

Reply:
My aim was to make it clear that this member function does not use any non-static member variables/functions. The only advantage that I can think of is that no "this" pointer is passed to the function, and as a result, the function is slightly more memory efficient, which is not that important. I'll change it back to a normal function.